pennylane/transforms/add_noise.py

```diff
@@ -0,0 +1,320 @@
+        ``noise_fns``, and queue it with :func:`~.pennylane.apply` as given below:
+
+        .. code-block:: python3
+
+            fcond3 = qml.noise.op_eq(qml.RX) & qml.noise.wires_in([0, 1])
+            def noise3(op, **kwargs):
+                qml.RZ(op.parameters[0] * 0.05, op.wires)
+                qml.apply(op) # <-- sandwiched between two RZs
+                qml.RZ(-op.parameters[0] * 0.05, op.wires)
+
+        Additionally, as noted before in the example above, one can provide keyword arguments for building the noise operations
+        based on the ``metadata`` based on properties like hardware topology, qubit relaxation times, etc.
+
+        .. code-block:: python3
+
+            fcond4 = qml.noise.op_in([qml.RX, qml.RZ])
+            def noise4(op, **kwargs):
+                qml.ThermalRelaxationError(op.parameters[0] * 0.5, kwargs["t1"],  kwargs["t2"], 0.6, op.wires)
+
+        Finally, one could build a noise model and combine it with another noise model via addition.
+
+        >>> noise_model = qml.NoiseModel({fcond1: noise1, fcond2: noise2, fcond3: noise3})
+        >>> noise_model += qml.NoiseModel({fcond4: noise4}, t1=2.0, t2=0.2)
+        >>> noise_model
+        NoiseModel({
+            OpIn(['PauliX', 'PauliY']): AmplitudeDamping(gamma=0.4)
+            BooleanFn(fcond2): PhaseDamping(gamma=0.9)
+            OpEq(RX) & WiresIn([0, 1]): noise3
+            OpIn(['RX', 'RZ']): noise4
+        }, t1 = 2.0, t2 = 0.2)
+
+        Now, one can use this ``noise_model`` for the transform as below:
+
+        .. code-block:: python3
+
+            def f(w, x, y, z):
+                qml.RX(w, wires=0)
+                qml.RY(x, wires=1)
+                qml.CNOT(wires=[0, 1])
+                qml.RY(y, wires=0)
+                qml.RX(z, wires=1)
+                return qml.expval(qml.Z(0) @ qml.Z(1))
+
+            dev = qml.device("default.mixed", wires=2)
+            qfunc_qnode = qml.QNode(f, dev)
+            noise_qnode = qml.transforms.add_noise(qfunc_qnode, noise_model)
+
+        and obtain the resulting modified qnode or device:
+
+        >>> print(qml.draw(qfunc_qnode, decimals=2)(0.1, 0.7, 0.8, 0.4))
+        0: ──RX(0.10)─╭●──RY(0.80)─┤ ╭<Z@Z>
+        1: ──RY(0.70)─╰X──RX(0.40)─┤ ╰<Z@Z>
+
+        >>> print(qml.draw(noise_qnode, decimals=2)(0.1, 0.7, 0.8, 0.4))
+        0: ──RZ(0.01)──RX(0.10)────────────RZ(-0.01)─╭●──RY(0.80)──PhaseDamping(0.90)────────────┤ ╭<Z@Z>
+        1: ──RY(0.70)──PhaseDamping(0.90)────────────╰X──RZ(0.02)──RX(0.40)────────────RZ(-0.02)─┤ ╰<Z@Z>
+
+        >>> noisy_dev = qml.transforms.add_noise(dev, noise_model)
+        >>> noisy_qfn = qml.QNode(f, noisy_dev)
+        >>> qml.math.isclose(noise_qnode(0.1, 0.7, 0.8, 0.4), noisy_qfn(0.1, 0.7, 0.8, 0.4))
+        True
+
+    .. details::
+        :title: Tranform Levels
+        :href: add-noise-levels
+
+        By default, ``add_noise`` transform will be added to the end of the "user" transforms,
```

Comment:
Nope. When used as a decorator, it gets applied in the order it is specified since, for the `QNode` it receives the "user" transforms will be the ones applied before it. For example - 

![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/cb1e0ad0-0895-4f9d-a34b-a36e4b60ed73)

Tweaked it a bit to make it more clear.